src/typescript/polymer.d.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+interface Constructor<T> {
+  new (...args: any[]): T;
+}
+
+/**
+ * An interface to match all Objects, but not primitives.
+ */
+interface Base {}
+
+/**
+ * A subclass-factory style mixin that extends `superclass` with a new subclass
+ * that implements the interface `M`.
+ */
+type Mixin<M> =
+    <C extends Base>(superclass: Constructor<C>) => Constructor<M & C>;
```

Comment:
Arg, we talked about this last week I think and I've already forgotten about the limitations of this technique. Or maybe it's a new technique?